### PR TITLE
Flow/v1

### PR DESCRIPTION
**Pull request by cryogenian**
[ch13042] 

This is extracts sink building machinery from destinations. I haven't come up with nice approach for `TempTableFlow` deduplication, it's either too brittle for usage in callsite or too generic and barely useful. Will return to this after Snowflake dest update. Although `Flow.delete` seems to be YAGNI case I'd say that we should have it and provide `().pure[ConnectionIO]` in destinations, to make flow intentions a bit clearer. 

`0.34.0-82cd35b` snapshotted

### CODE REVIEW

**Comment by cryogenian** [comment]
`0.34.0-da73128`

File: core/src/main/scala/quasar/lib/jdbc/destination/flow/implicits.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          s"Deleting ${idBatch.size} records"
+        case DataEvent.Commit(_) =>
+          "Commit"
+      }}
+      events.evalTap(logEvents).through(pipe.logStartAndEnd(logger))
+    }
+  }
+
+  implicit class StartEndLogger[F[_]: Sync, A, B](pipe: Pipe[F, A, B]) {
+    def logStartAndEnd(logger: Logger): Pipe[F, A, B] = { events =>
```
Comment (wemrysi) [inline core/src/main/scala/quasar/lib/jdbc/destination/flow/implicits.scala:48]: Might consider using `Stream#bracketCase` to a) always log termination and b) customize the end message based on the outcome.

File: core/src/main/scala/quasar/lib/jdbc/destination/flow/implicits.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      def trace(msg: => String) = Sync[F].delay(logger.trace(msg))
+      Stream.eval_(trace("Strating load")) ++
```
Comment (wemrysi) [inline core/src/main/scala/quasar/lib/jdbc/destination/flow/implicits.scala]: "Starting"

**Review by wemrysi — APPROVED** [review]
A few minor comments, but LGTM 👍 I think keeping `delete` is fine, I agree it conveys the intent better, even if not all implementations need it. One thing to note is that ignoring them will have consequences if we ever support semantic deletes in the event stream (i.e. a delete that doesn't correspond to an insert). Maybe we need an actual `Replace`/`Upsert` event to allow implementations to differentiate.

File: core/src/main/scala/quasar/lib/jdbc/destination/flow/implicits.scala (hunk 1 above)
Comment (cryogenian) [reply core/src/main/scala/quasar/lib/jdbc/destination/flow/implicits.scala:48]: :+1:

File: core/src/main/scala/quasar/lib/jdbc/destination/flow/implicits.scala (hunk 2 above)
Comment (cryogenian) [reply core/src/main/scala/quasar/lib/jdbc/destination/flow/implicits.scala]: Done

**Comment by cryogenian** [comment]
Yeah, I think that we should differentiate between semantic delete and prepare delete, when we had semantic delete :+1:

**Pull request merged**